cmd/agent/daemon/state/events_pipeline.go

```diff
@@ -88,20 +63,37 @@ func toProtoEvent(e *ebpftypes.Event) *castpb.Event {
 		event.EventType = castpb.EventType_EVENT_DNS
 
 		dnsEvent := args.Payload
-		dnsEvent.FlowDirection = convertFlowDirection(e.Context.ParseFlowDirection())
+		dnsEvent.FlowDirection = convertFlowDirection(e.Context.GetFlowDirection())
 		event.Data = &castpb.Event_Dns{
 			Dns: dnsEvent,
 		}
+
+		// Add dns cache.
+		for _, answ := range dnsEvent.Answers {
```

Comment:
This is sadly not enough. We need to trace the dns events per process, as different containers can have different DNS configs. I would suggest moving this logic into an enricher.

Reply:
Moving dns cache to per process will make it much less efficient. On each process restart cache is gone.

Maybe instead global cache value can be:
```go
type dnsCacheValue struct {
   // Record question and last pid from dns event. In most cases this should be a single record. 
   // If it contains more than 1 record during enrichment we would pick question which contains 
   // the same pid as event. 
   questions map[string]uint64 
}
```

Reply:
Added per cgroup cache